NPE in settings

<!-- Fill in the following form by adding your text below the explanation comments. -->
<!-- You can use the preview tab above to review your issue before submitting it. -->

## Bug description 
<!-- Enter a summarized description of what the bug/problem is, that you found -->
Just seen in a users logfile (probably independent from his problem, that Brouter is not working?!)

## Reproduce 
### Steps to reproduce the problem
<!-- Describe step-by-step how to reproduce the problem -->
Unkown

### Actual result after these steps 
<!-- Describe the actual issue/problem/behavior in detail, which happens after the steps above -->
Crash

### Expected result after these steps 
<!-- Describe what you expected to happen instead (correct behavior) -->
No crash

## c:geo version 
<!-- You will find the c:geo version in c:geo Menu -> About c:geo -->
2012.04.25

## Reproducible 
<!-- Yes / No (or describe under what conditions) -->
No

## System information 
<!-- Attach system information here if available (see c:geo Menu -> About c:geo -> Swipe right to System) -->
<!-- Keep the apostrophe at beginning and end to have it properly formatted -->


```
--- System information ---
c:geo version: 2021.04.25

Device:
-------
- Device type: ONEPLUS A3003 (OnePlus3, OnePlus)
- Available processors: 4
- Android version: 9
- Android build: ONEPLUS A3003_16_191104
- Sailfish OS detected: false
- Google Play services: enabled - 21.15.12 (100400-368160137)
- HW acceleration: enabled (default state)

Sensor and location:
-------
- Low power mode: active
- Compass capabilities: yes
- Rotation vector sensor: present
- Orientation sensor: present
- Magnetometer & Accelerometer sensor: present
- Direction sensor used: rotation vector

Program settings:
-------
- Hide caches: own/found
- Hide waypoints: -
- Set language: de_DE (system default)
- System date format: dd.MM.yy
- Debug mode active: no
- Live map mode: true
- OSM multi-threading: true / threads: 4
- Global filter: display all caches
- Last backup: never
- Routing mode: Straight
- Map: Bayern_oam.osm (Offline)
  - Id:
 cgeo.geocaching.maps.mapsforge.MapsforgeMapProvider$OfflineMapSource:primary:OSM-Maps/bayern_oam.osm.map
  - Atts: Maps: (c)Christian Kernbeis, Themes: (c)Tobias Kuehn, www.openandromaps.org 2012-2020
  - Theme: Elevate.zip:Elevate.xml

Services:
-------
- Geocaching sites enabled:
   geocaching.com: Logged in (Anmeldung OK) / PREMIUM
   opencaching.de: Logged in (Anmeldung OK)
- Geocaching.com date format: yyyy-MM-dd
- BRouter installed: true / connection available: false
- Installed c:geo plugins: contacts

Permissions & paths:
-------
- Fine location permission: granted
- Write external storage permission: granted
- System internal c:geo dir: /data/user/0/cgeo.geocaching (734,8 MB free) v2 internal isDir(10 entries)
- Legacy User storage c:geo dir: /storage/emulated/0/cgeo (634,8 MB free) v2 external non-removable isDir(5 entries)
- Geocache data: /storage/emulated/0/Android/data/cgeo.geocaching/files/GeocacheData (634,8 MB free) v2 external non-removable isDir(1711 entries)
- Internal theme sync (is turned off): /data/user/0/cgeo.geocaching/MapThemeData (734,8 MB free) v2 internal isDir(0 entries)
- Public Folders: #9
  - BASE: /cgeo (User-Defined)[/cgeo[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo, Av:true, files:>=4, dirs:>=5, totalFileSize:>=12,6 MB, free space: 634,8 MB, files on device: 3522560)
  - OFFLINE_MAPS: /OSM-Maps (User-Defined)[/OSM-Maps[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/primary%3AOSM-Maps::]] (Uri: content://com.android.externalstorage.documents/tree/primary%3AOSM-Maps/document/primary%3AOSM-Maps, Av:true, files:3, dirs:1, totalFileSize:773,4 MB, free space: 634,7 MB, files on device: 3522560)
  - OFFLINE_MAP_THEMES: /OSM-Maps/Themes (User-Defined)[/OSM-Maps/Themes[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/primary%3AOSM-Maps%2FThemes::]] (Uri: content://com.android.externalstorage.documents/tree/primary%3AOSM-Maps%2FThemes/document/primary%3AOSM-Maps%2FThemes, Av:true, files:2, dirs:0, totalFileSize:444,2 KB, free space: 634,7 MB, files on device: 3522560)
  - LOGFILES: /cgeo/logfiles (Default)[/cgeo/logfiles[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/logfiles]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Flogfiles, Av:true, files:1, dirs:0, totalFileSize:160,8 KB, free space: 634,6 MB, files on device: 3522560)
  - GPX: /cgeo/gpx (User-Defined)[/cgeo/gpx[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo%2Fgpx::]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo%2Fgpx/document/primary%3Acgeo%2Fgpx, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 634,6 MB, files on device: 3522560)
  - BACKUP: /cgeo/backup (Default)[/cgeo/backup[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/backup]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Fbackup, Av:true, files:1, dirs:0, totalFileSize:12,5 MB, free space: 634,6 MB, files on device: 3522560)
  - FIELD_NOTES: /cgeo/field-notes (Default)[/cgeo/field-notes[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/field-notes]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Ffield-notes, Av:true, files:2, dirs:0, totalFileSize:0 B, free space: 634,6 MB, files on device: 3522560)
  - SPOILER_IMAGES: /cgeo/GeocachePhotos (Default)[/cgeo/GeocachePhotos[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/GeocachePhotos]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2FGeocachePhotos, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 634,6 MB, files on device: 3522560)
  - TEST_FOLDER: [Legacy]/data/user/0/cgeo.geocaching/files/unittest (Default)[/data/user/0/cgeo.geocaching/files/unittest[FILE#1:p-file:///data/user/0/cgeo.geocaching/files::/unittest]] (Uri: file:///data/user/0/cgeo.geocaching/files/unittest, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 734,6 MB, files on device: -1)
- Map render theme path: Elevate.zip:Elevate.xml
- PersistedDocumentUris: #1
- TRACK: null
- Persisted Uri Permissions: #4
  - content://com.android.externalstorage.documents/tree/primary%3AOSM-Maps%2FThemes (3. Apr., 07:48):RW
  - content://com.android.externalstorage.documents/tree/primary%3AOSM-Maps (29. März, 07:31):RW
  - content://com.android.externalstorage.documents/tree/primary%3Acgeo (25. Apr., 09:45):RW
  - content://com.android.externalstorage.documents/tree/primary%3Acgeo%2Fgpx (3. Apr., 07:43):RW
- Database: /storage/emulated/0/Android/data/cgeo.geocaching/files/databases/data (v94, Size:46,8 MB) on user storage
-Settings: v5, Count:185
--- End of system information ---
```

## Additional context 
<!-- (optional, remove if not applicable) log files, reference to other similar issues, projects, sources, etc. -->
```
04-25 09:45:55.627 10665 10665 D AndroidRuntime: Shutting down VM
04-25 09:45:55.653 10665 10665 E AndroidRuntime: FATAL EXCEPTION: main
04-25 09:45:55.653 10665 10665 E AndroidRuntime: Process: cgeo.geocaching, PID: 10665
04-25 09:45:55.653 10665 10665 E AndroidRuntime: java.lang.RuntimeException: Unable to start activity ComponentInfo{cgeo.geocaching/cgeo.geocaching.settings.ViewSettingsActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String java.lang.Object.toString()' on a null object reference
04-25 09:45:55.653 10665 10665 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3037)
04-25 09:45:55.653 10665 10665 E AndroidRuntime: 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3172)
04-25 09:45:55.653 10665 10665 E AndroidRuntime: 	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:78)
04-25 09:45:55.653 10665 10665 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108)
04-25 09:45:55.653 10665 10665 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68)
04-25 09:45:55.653 10665 10665 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1906)
04-25 09:45:55.653 10665 10665 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
04-25 09:45:55.653 10665 10665 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:193)
04-25 09:45:55.653 10665 10665 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:6863)
04-25 09:45:55.653 10665 10665 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
04-25 09:45:55.653 10665 10665 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:537)
04-25 09:45:55.653 10665 10665 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
04-25 09:45:55.653 10665 10665 E AndroidRuntime: Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String java.lang.Object.toString()' on a null object reference
04-25 09:45:55.653 10665 10665 E AndroidRuntime: 	at cgeo.geocaching.settings.ViewSettingsActivity.onCreate(ViewSettingsActivity.java:76)
04-25 09:45:55.653 10665 10665 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:7149)
04-25 09:45:55.653 10665 10665 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:7140)
04-25 09:45:55.653 10665 10665 E AndroidRuntime: 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1288)
04-25 09:45:55.653 10665 10665 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3017)
04-25 09:45:55.653 10665 10665 E AndroidRuntime: 	... 11 more
```

--- a/main/src/cgeo/geocaching/settings/ViewSettingsActivity.java
+++ b/main/src/cgeo/geocaching/settings/ViewSettingsActivity.java
@@ -73,7 +73,9 @@ public class ViewSettingsActivity extends AbstractActivity {
             final Object value = entry.getValue();
             final String key = entry.getKey();
             final SettingsUtils.SettingsType type = getType(value);
-            items.add(new KeyValue(key, value.toString(), type));
+            if (value != null) { // should not happen, but...
+                items.add(new KeyValue(key, value.toString(), type));
+            }
         }
         Collections.sort(items, (o1, o2) -> o1.key.compareTo(o2.key));
 

